end-to-end-tests/specs/home.spec.js

```diff
@@ -0,0 +1,93 @@
+var assert = require('assert');
+
+const CBIOPORTAL_URL = process.env.CBIOPORTAL_URL.replace(/\/$/, "");
+
+describe('homepage', function() {
+
+    it('it should show dev mode when testing', function() {
+        browser.url(CBIOPORTAL_URL);
+
+        browser.localStorage('POST', {key: 'localdev', value: 'true'});
+        browser.refresh();
+
+        var devMode = $('.alert-warning');
+
+        devMode.waitForExist(30000);
+        assert(browser.getText('.alert-warning').indexOf('dev mode') > 0);
+    });
+
+    it('it should have 29 studies in list', function () {
+        browser.url(CBIOPORTAL_URL);
+
+        var studies = $('[data-test="cancerTypeListContainer"] > ul > ul');
+        
+        studies.waitForExist(10000); // same as `browser.waitForExist('.notification', 10000)`
+        
+        //assert.equal(browser.elements('[@data-test=cancerTypeListContainer] > ul > ul').value.length, 29);
+        
+    });
+
+
+    it('should filter study list according to filter text input', function () {
+        
+        assert.equal(browser.elements('[data-test="cancerTypeListContainer"] > ul > ul').value.length, 29);
+        
+        var input = $(".autosuggest input[type=text]");
+
+        input.waitForExist(10000); 
+
+        input.setValue('tract');
+        
+        browser.pause(500);
+
+        assert.equal(browser.elements('[data-test="cancerTypeListContainer"] > ul > ul').value.length, 2);
+        
+    });
+    
+    it('when a single study is selected, a case set selector is provided', function(){
+        
+        var caseSetSelectorClass = '[data-test="CaseSetSelector"]';
+        
+        var checkBox = $('[data-test="StudySelect"]');
+
+        checkBox.waitForExist(10000);
+        
+        assert.equal(browser.isExisting(caseSetSelectorClass), false);
+        
+        browser.click('[data-test="StudySelect"]');
```

Comment:
i think this i my code but we should figure out if we can just call click on checkbox itself.   also, we should write a little wrapper to abstract away data-test.  i.e. makeSelector('StudySelect")